stdlib/public/core/StringCharacterView.swift

```diff
@@ -739,6 +741,12 @@ extension String.CharacterView {
   ///
   /// - Complexity: O(*n*) if the underlying string is bridged from
   ///   Objective-C, where *n* is the length of the string; otherwise, O(1).
+  @available(swift, introduced: 4.0)
+  public subscript(bounds: Range<Index>) -> Substring.CharacterView {
```

Comment:
What's the story around substrings and bridging? Do we automatically bridge an objc string when it's substring'd, or can a substring be `NSString`-backed?